pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -409,11 +422,35 @@ private File downloadSegmentFromDeepStore(String segmentName, SegmentZKMetadata
       throws Exception {
     File tempRootDir = getTmpSegmentDataDir("tmp-" + segmentName + "-" + UUID.randomUUID());
     FileUtils.forceMkdir(tempRootDir);
+    if (_isStreamSegmentDownloadUntar && zkMetadata.getCrypterName() == null) {
```

Comment:
Does that mean we cannot stream & untar if the segment is encrypted? If so, specify it to one comment in this method here.

Reply:
Also, can this if statement be put into the try block, so that you don't have to duplicate the code for deleting tempRootDir.